syncapi/sync/requestpool.go

```diff
@@ -54,17 +57,27 @@ func (rp *RequestPool) OnIncomingSyncRequest(req *http.Request, device *authtype
 			JSON: jsonerror.Unknown(err.Error()),
 		}
 	}
+
 	logger := util.GetLogger(req.Context()).WithFields(log.Fields{
-		"userID":   device.UserID,
-		"deviceID": device.ID,
-		"since":    syncReq.since,
-		"timeout":  syncReq.timeout,
-		"limit":    syncReq.limit,
+		"user_id":   device.UserID,
+		"device_id": device.ID,
+		"since":     syncReq.since,
+		"timeout":   syncReq.timeout,
+		"limit":     syncReq.limit,
 	})
 
 	currPos := rp.notifier.CurrentPosition()
 
-	if shouldReturnImmediately(syncReq) {
+	returnImmediately := shouldReturnImmediately(syncReq)
+	if !returnImmediately {
+		if waiting, werr := rp.db.SendToDeviceUpdatesWaiting(
```

Comment:
It feels like this check should be in `shouldReturnImmediately` so we don't have this strange control flow where we then check additional places.